src/BTVNanoCommissioning/utils/correction.py

```diff
@@ -160,71 +291,128 @@ def load_pu(campaign, path):
 
 ## BTag SFs
 def load_BTV(campaign, path, tagger):
-    _btag_path = f"BTVNanoCommissioning.data.BTV.{campaign}"
-    if tagger == "DeepCSVB" or tagger == "DeepJetB":
-        with importlib.resources.path(_btag_path, path[tagger]) as filename:
-            deepsf = BTagScaleFactor(
-                filename,
-                BTagScaleFactor.RESHAPE,
-                methods="iterativefit,iterativefit,iterativefit",
-            )
-    elif tagger == "DeepCSVC" or tagger == "DeepJetC":
-        deepsf = "BTVNanoCommissioning/data/BTV/" + campaign + "/" + path[tagger]
-    return deepsf
+    if campaign == "Rereco17_94X":
+        _btag_path = f"BTVNanoCommissioning.data.BTV.{campaign}"
+        if tagger == "DeepCSVB" or tagger == "DeepJetB":
+            with importlib.resources.path(_btag_path, path[tagger]) as filename:
+                deepsf = BTagScaleFactor(
+                    filename,
+                    BTagScaleFactor.RESHAPE,
+                    methods="iterativefit,iterativefit,iterativefit",
+                )
+        elif tagger == "DeepCSVC" or tagger == "DeepJetC":
+            deepsf = "BTVNanoCommissioning/data/BTV/" + campaign + "/" + path[tagger]
+        return deepsf
 
 
-### Lepton SFs
-def eleSFs(ele, campaign, path):
-    _ele_path = f"BTVNanoCommissioning.data.LSF.{campaign}"
-    ext = extractor()
-    with contextlib.ExitStack() as stack:
-        real_paths = [
-            stack.enter_context(importlib.resources.path(_ele_path, f))
-            for f in path.values()
-        ]
-        ext.add_weight_sets(
-            [
-                f"{paths} {file}"
-                for paths, file in zip(path.keys(), real_paths)
-                if "ele" in paths
-            ]
+def puwei(correct_map, nPU):
+    if "correctionlib" in str(type(correct_map["PU"])):
+        return correct_map["PU"][list(correct_map["PU"].keys())[0]].evaluate(
+            nPU, "nominal"
```

Comment:
similar question as for load_BTV function

Reply:
This has nothing to do with load function, it returns the pu weights , similar as the btagSFs, eleSFs, muSF

Reply:
Thx for clarification!